src/controllers/hid/hidcontroller.cpp

```diff
@@ -137,6 +137,70 @@ int HidController::close() {
     return 0;
 }
 
+void HidController::processInputReport(int& bytesRead, const int& currentBufferIndex) {
+    Trace process("HidController processInputReport");
+    unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+    unsigned char* pCurrentBuffer = m_pPollData[currentBufferIndex];
+    // Some controllers such as the Gemini GMX continuously send input reports even if it
+    // is identical to the previous send input report. If this loop processed all those redundant
+    // input report, it would be a big performance problem to run JS code for every  input report and
+    // would be unnecessary.
+    // This assumes that the redundant input report all use the same report ID. In practice we
+    // have not encountered any controllers that send redundant input report with different report
+    // IDs. If any such devices exist, this may be changed to use a separate buffer to store
+    // the last input report for each report ID.
+    if (bytesRead == m_iLastPollSize &&
+            memcmp(pCurrentBuffer, pPreviousBuffer, bytesRead) == 0) {
+        return;
+    }
+    m_iLastPollSize = bytesRead;
+    m_iPollingBufferIndex = currentBufferIndex;
+    auto incomingData = QByteArray::fromRawData(
+            reinterpret_cast<char*>(pCurrentBuffer), bytesRead);
+    receive(incomingData, mixxx::Time::elapsed());
```

Comment:
The time should be measured as early as possible to a void a jitter do to suspends in the code before. The best would be to ask hidapi for it. But it looks like that is not implemented, is it?

Reply:
By the way, we have no prefix rule for i.

Reply:
I removed the prefix i !
I'm not sure about what should be changed on the time measurement. I didn't wrote this code I just moved it on the request from uklotze above, to get rid of the code duplication. What should be measured here - the execution time of the JavaScript call which is implemented in receive, or the execution time of HIDAPI, which is called outside of this function?